schemas/org/openstax/ec/accessed_studyguide/v1/avro_builder.rb

```diff
@@ -2,12 +2,14 @@
 
 import 'types/uuid'
 import 'types/date_time'
+import 'types/device'
+import 'types/ordered_session'
 
 record :accessed_studyguide_v1 do
+  extends :device
+  extends :ordered_session
   required :user_uuid, :uuid
```

Comment:
What about `extend :whoami` and have that include the user ID and the device ID?

Note that user ID _may_ not always be required, e.g. what if they let you use the study guide in a chapter without logging in but then have you sign up / log in after.

Reply:
@jpslav what if we require the user_uuid (and default it to 'xxxxxx' if not logged in)